src/lib.rs

```diff
@@ -835,6 +835,27 @@ impl<T: TensorType> Tensor<T> {
         }
     }
 
+    /// Sets (copies) the tensor values to the provided ones.
+    /// 
+    /// ```
+    /// # use tensorflow::Tensor;
+    /// let a = Tensor::new(&[2, 2]).with_values(&[0_i32, 1, 2, 3]);
+    /// ```
+    ///
+    /// # Panic
+    /// If `values.len() != self.len()`.
+    pub fn with_values(mut self, values: &[T]) -> Self {
+        assert!(
+            self.len() == values.len(), 
+            "length of values array ({}) is not equal to tensor total elements ({})", 
+            values.len(), self.len()
+        );
+        for i in 0..self.len() {
+            self[i] = values[i].clone();
+        }
```

Comment:
We can make this loop more idiomatic. Can you try this instead?

```rust
for (e, v) in self.iter_mut().zip(values) {
    *e = v.clone();
}
```

Reply:
What about [`clone_from`](https://doc.rust-lang.org/std/clone/trait.Clone.html#method.clone_from) instead?

Reply:
@sanmai-NL Yes, good thinking. `e.clone_from(v)` should work here as well.